### PR TITLE
Switch server-side mapimg to Qt

### PR DESCRIPTION
**Pull request by lmoureaux**
The server-side map image code was using a handcrafted ppm writer. This produces huge files. Since we have Qt with many built-in image writers, use that instead. This also changes the default to png since it's just a better format; and replaces the black background with transparent.

### CODE REVIEW

File: cmake/FreecivInstall.cmake [hunk 1]
```diff
@@ -22,12 +22,10 @@ install(
   ${CMAKE_SOURCE_DIR}/dist/licenses/FTL.txt
   ${CMAKE_SOURCE_DIR}/dist/licenses/GPL2.txt
   ${CMAKE_SOURCE_DIR}/dist/licenses/GPL3.txt
-  ${CMAKE_SOURCE_DIR}/dist/licenses/IMAGEMAGICK.txt
```
Comment (jwrober) [inline cmake/FreecivInstall.cmake:25]: I think we should delete this file too.

**Review by jwrober — APPROVED** [review]

**Comment by jwrober** [comment]
You also need to update the github action workflow files and readme to remove the dependency.

**Review by jwrober — CHANGES_REQUESTED** [review]
See comments

File: cmake/FreecivInstall.cmake (hunk 1 above)
Comment (lmoureaux) [reply cmake/FreecivInstall.cmake:25]: My bad, I missed that there were two licenses for ImageMagick

**Comment by lmoureaux** [comment]
> You also need to update the github action workflow files and readme to remove the dependency.

I don't see it there - we never supported building with ImageMagick, I think the mention in the docs was a leftover from upstream.

**Comment by jwrober** [comment]
> > You also need to update the github action workflow files and readme to remove the dependency.
> 
> I don't see it there - we never supported building with ImageMagick, I think the mention in the docs was a leftover from upstream.

I am referring to the installation of a library we no long need that is referred to in the README and the gh actions build files. You removed it from docs, but it still remains in other places.

**Comment by lmoureaux** [comment]
I don't understand... this is the full list of packages we install on Ubuntu:
https://github.com/longturn/freeciv21/blob/572b15e8d00bc3446155026b4430a8651e0211a6/.github/workflows/build.yaml#L43-L55

And on Windows:
https://github.com/longturn/freeciv21/blob/572b15e8d00bc3446155026b4430a8651e0211a6/.github/workflows/build.yaml#L140-L151

What do you want me to remove?

**Comment by jwrober** [comment]
Ok I stand corrected. I guess we had a dep install in docs that we didn't do anywhere else.

**Review by jwrober — APPROVED** [review]

**Pull request merged**